include/envoy/router/router.h

```diff
@@ -130,10 +130,26 @@ class CorsPolicy {
    */
   virtual const absl::optional<bool>& allowCredentials() const PURE;
 
+  /**
+   * @return bool Whether CORS policy uses filterEnabled instead of the deprecated enabled.
+   */
+  virtual bool hasFilterEnabled() const PURE;
```

Comment:
Can we potentially just hide all of this behind the interface? Maybe just `enabled()` and `shadowEnabled()` and the rest is an impl detail?